addons/pos_sale/migrations/13.0.1.0/post-migration.py

```diff
@@ -3,6 +3,17 @@
 from openupgradelib import openupgrade
 
 
+def _map_crm_team_id(env):
+    """From now on, every pos.order will have stored its crm.team relation, so if it's
+    changed later, those sales don't go to the new team. Anyway we have to map it in
+    the old orders so we don't loose that info in the sales report."""
+    po_sale = env["pos.order"].search([("crm_team_id", "=", False)])
+    for config in po_sale.config_id.filtered("crm_team_id"):
+        po_sale.filtered(
+            lambda x: x.config_id == config
+        ).crm_team_id = config.crm_team_id
```

Comment:
Always use SQL when possible, as you don't know if a custom is overwriting `write` method for doing a heavy check and in a certain DB times go upper exponentially:
```suggestion
    openupgrade.logged_query(
        env.cr,
        """
        UPDATE pos_order po
        SET crm_team_id = pc.crm_team_id
        FROM pos_config pc
        WHERE pc.id = po.config_id
            AND po.crm_team_id IS NULL
            AND pc.crm_tema_id IS NOT NULL
        """,
    )
```

Reply:
Done :+1: